src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -164,36 +164,38 @@ class SidebarLinks extends React.Component {
 SidebarLinks.propTypes = propTypes;
 SidebarLinks.defaultProps = defaultProps;
 
-export default compose(
-    withLocalize,
-    withCurrentUserPersonalDetails,
-    withWindowDimensions,
-    withOnyx({
+export default React.memo(
```

Comment:
Additionally, can you please leave a comment why to use this, basically another question woul dbe why to use it here and not at other places

Reply:
Yeah so, `SidebarLinks` is rendered in `BaseSidebarScreen`, which also renders the `PopoverMenu`. So when the `isCreateMenuActive` state changes it rerenders the `PopoverMenu` which is desired, but also the `SidebarLinks`, which doesn't need to rerender:

https://github.com/Expensify/App/blob/6db559f9b42dbfa5e1e09e60d692272d85039e9c/src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js#L112-L129

It's important to wrap the Function returned by `compose`. This way none of the HoCs has to run. Especially `withOnyx` is expensive to run. That's why we don't just turn e.g. the class to a `PureComponent`, or wrap the Class with `React.memo`, but really wrap the whole thing

Reply:
Another solution here could be to move the `PopoverMenu` with the `isVisible` state into its own component, so only that component rerenders and not the whole screen.

Reply:
Oh yeah, thank you for this, I mean to add a comment to the code for any engineer wondering.

Also I feel like reading about the React memo, what is curious to me is why not to use it in more places? Or did you not find other place which would have this rendering issue?

Reply:
@mountiny `React.memo` should be used pretty sparingly because it can have some pretty unintentional side-effects. It can be a bit lazy to just use it everywhere, thinking it'll be a good fix. I believe, primarily, it only does a shallow comparison of props, which isn't always fully understood.

> Another solution here could be to move the PopoverMenu with the isVisible state into its own component, so only that component rerenders and not the whole screen.

I'd actually much rather do this than use `React.memo` in this PR. I think a lot of the sidebar re-rendering problems will go away when I am able to apply the Onyx property selectors to it (see more details of what this is about in [this GH](https://github.com/Expensify/Expensify/issues/222430)).

Reply:
Great, yeah, let's definitely make new PRs for each refactor.